tooling/nargo_cli/src/cli/dap_cmd.rs

```diff
@@ -182,11 +211,40 @@ fn loop_uninitialized_dap<R: Read, W: Write>(
     Ok(())
 }
 
+fn run_preflight_check(backend: &Backend, args: DapCommand) -> Result<(), DapError> {
+    let project_folder = if let Some(project_folder) = args.preflight_project_folder {
+        project_folder
+    } else {
+        return Err(DapError::PreFlightGenericError("Noir Debugger could not initialize because the IDE (for example, VS Code) did not specify a project folder to debug.".into()));
+    };
+
+    let package = args.preflight_package.as_deref();
+    let prover_name = args.preflight_prover_name.as_deref().unwrap_or(PROVER_INPUT_FILE);
+
+    let generate_acir = args.preflight_generate_acir.unwrap_or(false);
+    let skip_instrumentation = args.preflight_skip_instrumentation.unwrap_or(false);
+
+    let _ = load_and_compile_project(
+        backend,
+        project_folder.as_str(),
+        package,
+        prover_name,
+        generate_acir,
+        skip_instrumentation,
+    )?;
+
+    Ok(())
+}
+
 pub(crate) fn run(
     backend: &Backend,
-    _args: DapCommand,
+    args: DapCommand,
     _config: NargoConfig,
 ) -> Result<(), CliError> {
+    if args.preflight_check {
+        return run_preflight_check(backend, args).map_err(CliError::DapError);
+    }
+
```

Comment:
I would leave a comment here explaining why the preflight checks are necessary.

Reply:
Great point, on it.